packages/design-system/src/components/contextMenu/menu.js

```diff
@@ -102,178 +103,188 @@ function getFocusableChildren(parent, isSubMenu) {
-}) => {
-  const { isRTL } = props;
-  const { focusedId, isIconMenu, isHorizontal, onDismiss, setFocusedId } =
-    useContextMenu(({ state, actions }) => ({
-      focusedId: state.focusedId,
-      isIconMenu: state.isIconMenu,
-      isHorizontal: state.isHorizontal,
-      onDismiss: actions.onDismiss,
-      setFocusedId: actions.setFocusedId,
-    }));
-  const mouseDownOutsideRef = useMouseDownOutsideRef(() => {
-    isOpen && !isSubMenu && onDismiss();
-  });
-  const menuRef = useRef(null);
-  const composedListRef = useCombinedRefs(mouseDownOutsideRef, menuRef);
-
-  /**
-   * Focus the first element when the user focuses the wrapper
-   * with their keyboard.
-   *
-   * Clicking in the wrapper should not automatically focus the first
-   * focusable element.
-   */
-  const handleFocus = useCallback(
-    (evt) => {
-      onFocus(evt);
-
-      if (
-        menuRef.current === evt.target &&
-        !menuRef.current.contains(evt.target)
-      ) {
-        const focusableChildren = getFocusableChildren(menuRef.current);
-        focusableChildren?.[0]?.focus();
-      }
+const Menu = forwardRef(
+  (
+    {
+      children,
+      disableControlledTabNavigation,
+      isOpen,
+      onFocus = noop,
+      isSubMenu = false,
+      isSecondary = false,
+      parentMenuRef,
+      onCloseSubMenu = noop,
+      dismissOnEscape = true,
+      ...props
     },
-    [onFocus]
-  );
+    ref
+  ) => {
+    const { isRTL } = props;
+    const { focusedId, isIconMenu, isHorizontal, onDismiss, setFocusedId } =
+      useContextMenu(({ state, actions }) => ({
+        focusedId: state.focusedId,
+        isIconMenu: state.isIconMenu,
+        isHorizontal: state.isHorizontal,
+        onDismiss: actions.onDismiss,
+        setFocusedId: actions.setFocusedId,
+      }));
+    const mouseDownOutsideRef = useMouseDownOutsideRef(() => {
+      isOpen && !isSubMenu && onDismiss();
+    });
+    const menuRef = useRef(null);
+    const composedListRef = useCombinedRefs(mouseDownOutsideRef, menuRef, ref);
 
-  /**
-   * Allow navigation of the list using the UP and DOWN arrow keys.
-   * Allow navigation between the parent menu and submenu with LEFT and RIGHT arrow keys.
-   * Close menu if ESCAPE is pressed.
-   *
-   * @param {Event} event The synthetic event
-   * @return {void} void
-   */
-  const handleKeyboardNav = useCallback(
-    (evt) => {
-      const { key } = evt;
-      if (key === 'Escape') {
-        onDismiss(evt);
-        return;
-      }
+    /**
+     * Focus the first element when the user focuses the wrapper
+     * with their keyboard.
+     *
+     * Clicking in the wrapper should not automatically focus the first
+     * focusable element.
+     */
+    const handleFocus = useCallback(
+      (evt) => {
+        onFocus(evt);
+        const menuChildren = menuRef.current.children || [];
+        if (
+          menuRef.current === evt.target &&
+          ![...menuChildren].some((child) => child.contains(evt.target))
```

Comment:
🤔 for an empty array, doesn't `![...menuChildren].some((child) => child.contains(evt.target))` come out as `true`?

<img width="223" alt="image" src="https://user-images.githubusercontent.com/22185279/162756115-582f3c3f-c058-4812-8c40-819efaddc4b1.png">

I guess it doesn't matter too much since there's no focusable children? Maybe there should be a length check to make sure that the rest of that logic isn't run?
```suggestion
          menuChildren.length && ![...menuChildren].some((child) => child.contains(evt.target))
```

Reply:
ah good catch. thanks